controllers/cloud.redhat.com/clowdenvironment_controller.go

```diff
@@ -54,15 +56,34 @@ func (r *ClowdenvironmentReconciler) Reconcile(ctx context.Context, req ctrl.Req
 	)
 
 	if ready, _ := VerifyClowdEnvReady(env); ready {
-		ns := env.Spec.TargetNamespace
-		r.Log.Info("Clowdenvironment ready", "ns-name", ns)
+		nsName := env.Spec.TargetNamespace
+		r.Log.Info("Clowdenvironment ready", "ns-name", nsName)
 
-		if err := CreateFrontendEnv(ctx, r.Client, ns, env); err != nil {
-			r.Log.Error(err, "Error encountered with frontend environment", "ns-name", ns)
-			UpdateAnnotations(ctx, r.Client, map[string]string{"env-status": "error"}, ns)
+		if err := CreateFrontendEnv(ctx, r.Client, nsName, env); err != nil {
+			r.Log.Error(err, "Error encountered with frontend environment", "ns-name", nsName)
+			UpdateAnnotations(ctx, r.Client, map[string]string{"env-status": "error", "status": "error"}, nsName)
```

Comment:
It feels kinda wrong to use annotations like this. Isn't this what status and conditions are for? If you have been using this pattern and elsewhere, and if others have signed off on it - cool. I don't want to tell you to go change it. But I wanted to bring it up just in case its worth thinking about later.

Reply:
Going to come back to this is a future PR.